content/blog/kubernetes-ingress-with-aws-alb-ingress-controller-and-pulumi-crosswalk/index.md

```diff
@@ -0,0 +1,255 @@
+    deployDashboard   : false,
+    vpcCniOptions     : {
+    warmIpTarget    : 4,
+    },
+});
+
+export const clusterName = cluster.eksCluster.name;
+export const kubeconfig = cluster.kubeconfig;
+export const clusterNodeInstanceRoleName = cluster.instanceRoles.apply(roles => roles[0].name);
+export const nodesubnetId = cluster.core.subnetIds;
+```
+
+Configure the Public subnets in the console as defined in
+[this guide](https://kubernetes-sigs.github.io/aws-alb-ingress-controller/guide/controller/config/#subnet-auto-discovery).  
+
+## Step 3: Deploy AWS ALB Ingress Controller
+
+Lets confirm that the EKS cluster is up using the following commands:
+
+    $ pulumi stack export kubeconfig > kubeconfig.yaml
+    $ export KUBECONFIG=kubeconfig.yaml
+    $ kubectl get nodes
+    NAME                          STATUS   ROLES    AGE    VERSION
+    ip-10-10-0-58.ec2.internal    Ready    <none>   7h8m   v1.12.7
+    ip-10-10-1-167.ec2.internal   Ready    <none>   7h8m   v1.12.7
+    ip-10-10-1-84.ec2.internal    Ready    <none>   7h8m   v1.12.7
+
+Adequate roles and policies must be configured in AWS and available to
+the node(s) running the controller. How access is granted is up to you.
+Some will attach the needed rights to node's role in AWS. Others will
+use projects like [kube2iam](https://github.com/jtblin/kube2iam). We
+attach a minimal IAM policy to the EKS worker nodes and then declare
+this on the EKS cluster as shown in the code below.
+
+When declaring the ALB Ingress controller we simply re-use the Helm
+chart as part of the code. There is no need to rewrite all the logic or
+install Tiller in the EKS cluster. This frees you from thinking about
+RBAC for Helm, Tiller and the k8s cluster per se'.
+
+With the default "instance mode" Ingress traffic starts from the ALB and
+reaches the
+[NodePort](https://kubernetes.io/docs/concepts/services-networking/service/#nodeport)
+opened for the service. Traffic is then routed to the container Pods
+within cluster. This is all encoded using Pulumi libraries below. If you
+wish to use "ip-mode" with your Ingress such that traffic directly
+reaches your pods, you will need to modify the
+`alb.ingress.kubernetes.io/target-type` annotation when using the helm
+chart.
+
+Append `index.ts` file from Step 1 with the code below and run
+`pulumi up`:
+
+```typescript
+// STEP 2: Declare ALB Ingress Controller from a Helm Chart
+const albingresscntlr = new k8s.helm.v2.Chart("albingresscontroller", {
+    chart: "http://storage.googleapis.com/kubernetes-charts-incubator/aws-alb-ingress-controller-0.1.9.tgz",
+    values: {
+        clusterName: clusterName,
+        autoDiscoverAwsRegion: "true",
+        autoDiscoverAwsVpcID: "true",
+        name: "alb-ingress-cntroller-0.1.9",
+        namespace: "kube-system",
+    },
+}, { provider: cluster.provider });
+```
+
+Confirm the alb-ingress-controller was created as follows:
+
+    $ kubectl get pods -n default | egrep -o alb-ingress[a-zA-Z0-9-]+
```

Comment:
replace lines 165-166 with the following:

```bash
$ kubectl get pods -n default | grep alb
alb-aws-alb-ingress-controller-58f44d4bb8lxs6w

$ kubectl logs alb-ingress-controller-58f44d4bb8lxs6w
-------------------------------------------------------------------------------
AWS ALB Ingress controller
  Release:    v1.1.2
  Build:      git-cc1c5971
  Repository: https://github.com/kubernetes-sigs/aws-alb-ingress-controller.git
-------------------------------------------------------------------------------
```

Reply:
Done